### DISCUSSION TITLE
How to Use a Custom Antenna in Simulation Model?

### DISCUSSION THREAD

**Discussion by cyasing** [NanoComp/meep#2300] [Q&A]
I have created my own microstrip antenna in Ansys, and would like to import it into the system setup for meep (microwave imaging of brain). The region of simulation will be in near-field, and there will be multiple antennas (one transmitter, others receiver).

Is it possible to maybe import a Huygen's box (around the antenna) so that the equivalent current distributions on the box can be modeled as multiple point sources (instead of the antenna)? Or can we just import an antenna design to the simulation region?

**Comment by dalarev** [comment]
I think the [docs ](https://meep.readthedocs.io/en/latest/) have some notes on this, but you can also read [this](https://github.com/NanoComp/meep/discussions/2218) relevant discussion.

**Reply by cyasing** [reply]
Excellent! I'll go through the discussion as soon as I can.

**Reply by cyasing** [reply]
@dalarev Could you elaborate on the parsing for field ndarray? I'm not able to understand how the size of field specified in the docstring relates to having all 6 components.

**Reply by dalarev** [reply]
Sure: in the case of 3D, the source plane will have one `= 1` component. In my case, the source plane is on the _xz_ plane, so `Ny = 1`, and `size(field) = (6, Nx, 1, Nz)`.

If you're dealing with ANSYS fld files, the formatting may be like: `X Y Z complex(Ex) complex(Ey) complex(Ez)`

This is where I was getting confused, because for [this](https://github.com/NanoComp/meep/discussions/2218#discussioncomment-3941180) function, it seems like you have to think about it as points on the grid / plane.

You could do this with [pandas](https://stackoverflow.com/questions/65698111/python-pandas-transform-x-y-list-to-x-y-coordinate-system), but I ended up just doing it myself. From your fld, pull and define your `x, y, z` arrays and find the unique values: `Nx = len(np.unique(x))`. Then you can reshape your data to `(Nx, Nz)`, in this example.

Maybe best to continue further discussion in that thread. I'm still working through it as I haven't been able to run a simulation successfully yet.

Hope that helps.

**Reply by cyasing** [reply]
@dalarev Okay, I understand this. I'll continue on that thread if needed.
